index.bs

```diff
@@ -245,6 +245,30 @@ partial interface NeuralNetworkContext {
 };
 </script>
 
+### concat ### {#api-neuralnetworkcontext-concat}
+<script type=idl>
+partial interface NeuralNetworkContext {
+  Operand concat(sequence<Operand> inputs, long dimension);
+};
+</script>
+
+<div algorithm=concat>
+    **Arguments:**
+        - *inputs*: a sequence of {{Operand}}. All input tensors must have the
+            same shape, except for the size of the dimension to concatenate on.
+        - *dimension*: a {{long}}. It specifies the dimension to concatenate on.
+            Its value is in the interval [0, N) where N is the rank of input
+            tensors.
+        
+    **Returns:** an {{Operand}}. The output tensor is concatenated from all
+    input tensors along the *dimension*. The output tensor has the same shape of
+    the input tensors, except for the concatenation dimension. For the
+    concatenation dimension of the output tensor, its size is the sum of size
+    along that dimension of all input tensors.
+
+    Concatenates the input tensors along the given dimension.
+</div>
```

Comment:
Suggestion: "The concatenated tensor of all the inputs along the `axis`. The output tensor has the same shape except on the dimension that all the inputs concatenated along."

Reply:
Your suggestion sounds good. I'll incorporate it. Thanks.

Reply:
Fixed in the new commit. Please take a look.

Reply:
Looks good to me. One approach common in many frameworks is the use of negative axis values to count backwards from the last axis. Thus, axis=-1 would mean last axis, and axis=-2 would mean second axis from last. This may be worth adopting, as it will simplify translation from other frameworks to this. What do you think?

Reply:
Thanks @gramalingam for proposing the negative axis values. AFAIK, there are some OS ML APIs, e.g. [join operator of DirectML](https://docs.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_join_operator_desc) and [concatenation operation of NNAPI](https://developer.android.com/ndk/reference/group/neural-networks#group___neural_networks_1ggaabbe492c60331b13038e39d4207940e0a44cbea825c4b224dd3ea757e9b1f65ed), don't support the negative axis values. If we decided to support that, it would increase the complexity of browser implementation for those APIs. Instead I would prefer to leaving it to JavaScript frameworks. Does it make sense?

Reply:
@gramalingam, are you fine with non supporting negative axis values in this PR?

Reply:
It is fine to merge this PR as is. (Note that one situation where the graph-builder, say javascript frameworks, cannot do this statically is when the rank of the tensors is not known statically.)

Reply:
Thanks @gramalingam for pointing out this situation. As this is not required by [first wave models](https://github.com/webmachinelearning/webnn/blob/master/op_compatibility/first_wave_models.md), let's merge it. If there are new requirement, we can create PR to amend it.